addons/mail/static/src/core/common/chat_bubble.xml

```diff
@@ -42,8 +42,8 @@
         <t t-if="message.isSelfAuthored">
             <i class="fa fa-mail-reply me-1 opacity-75"/>You:
         </t>
-        <t t-elif="!message.author?.eq(thread.correspondent?.persona)">
-            <t t-esc="message.author?.name ?? message.email_from"/>:
+        <t t-elif="!(message.author_id || message.author_guest_id)?.eq(thread.correspondent?.persona)">
+            <t t-esc="message.authorName"/>:
```

Comment:
Note: This is likely preferable to using the name directly, as it properly accounts for overrides.